driver/driver.go

```diff
@@ -418,9 +418,24 @@ func (d *ContrailDriver) DeleteEndpoint(req *network.DeleteEndpointRequest) erro
 	// containerID := req.Options["vmname"]
 	containerID := req.EndpointID
 
-	// TODO JW-429
-	// ifaces = contrailInstance.GetVirtualMachineInterfaces()
-	// agent.DeletePort(contrailVif.GetUuid())
+	meta, err := d.networkMetaFromDockerNetwork(req.NetworkID)
+	if err != nil {
+		return err
+	}
+
+	contrailNetwork, err := d.controller.GetNetwork(meta.tenant, meta.network)
+	log.Infoln("Retreived Contrail network:", contrailNetwork.GetUuid())
+	if err != nil {
+		return err
+	}
```

Comment:
```
contrailNetwork, err := d.controller.GetNetwork(meta.tenant, meta.network)
 +	log.Infoln("Retreived Contrail network:", contrailNetwork.GetUuid())
 +	if err != nil {
 +		return err
 +	}
```

I think logic here is faulty (and in all other places in this file as well). `contrailNetwork` could be nil, but we use `contrailNewtork.GetUuid()`, which would cause nil pointer dereference... If anything, we should first check for error, then log. 

This is a bug I'm pretty sure, which may explain some crashes we've been getting lately.

`controller.GetNetwork` uses `types.VirtualNetworkByName`, which uses `FindByName`, which can return nil:

```
func (c *Client) FindByName(typename string, fqn string) (IObject, error) {
	uuid, err := c.UuidByName(typename, fqn)
	if err != nil {
		return nil, err
	}
	href := fmt.Sprintf(
		"http://%s:%d/%s/%s", c.server, c.port, typename, uuid)
	return c.readObject(typename, href)
}
```

I've created a ticket to fix this throughout the driver code (JW-1113). For now, I think we can fix it only here.